test/message-stream.ts

```diff
@@ -338,7 +344,7 @@ describe('MessageStream', () => {
         const fakeResponses = [{}, {}, {}, {}, {}];
         const recieved: object[] = [];
 
-        messageStream.on('data', chunk => recieved.push(chunk))
+        messageStream.on('data', chunk => recieved.push(chunk as object))
```

Comment:
Seems like if we can avoid casting emitted data we should. Does implementing an interface that lists the correct types work here?

```typescript
interface StreamEvents {
  on(event: 'data', listener: (data: {}) => void): void;
}

class MessageStream implements StreamEvents {}
```

?